nion/swift/LinePlotCanvasItem.py

```diff
@@ -430,31 +430,46 @@ def __view_to_intervals(self, data_and_metadata: DataAndMetadata.DataAndMetadata
         else:
             left = right = 0.0
 
-        data_size = int(data_and_metadata.data_shape[-1])
-        left_channel = int(numpy.floor(left * data_size))
-        right_channel = int(numpy.ceil(right * data_size))
-        if left_channel > right_channel:
-            right_channel, left_channel = left_channel, right_channel
-        if left_channel not in range(data_size):
-            left_channel = 0
-        if right_channel not in range(data_size):
-            right_channel = data_size
-
-        channel_data = data_and_metadata.data
-        assert channel_data is not None
-        if channel_data[..., left_channel:right_channel].size > 0:
-            data_min = numpy.amin(channel_data[..., left_channel:right_channel])
-            data_max = numpy.amax(channel_data[..., left_channel:right_channel])
-        else:
-            data_min = numpy.amin(channel_data[..., :])
-            data_max = numpy.amax(channel_data[..., :])
+        top_level_data_size = int(data_and_metadata.data_shape[-1])
+        top_level_offset = data_and_metadata.intensity_calibration.offset
+        top_level_scale = data_and_metadata.intensity_calibration.scale
+
+        y_minimums = []
+        y_maximums = []
+
+        for display_value in self.__display_values_list:
```

Comment:
It's better to just use the `__xdata_list` directly rather than asking for the adjusted data values. The `__xdata_list` is the actual drawn data at any given moment.